android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/ConfigurationRegistry.kt

```diff
@@ -320,15 +321,46 @@ constructor(
             composition
               .retrieveCompositionSections()
               .groupBy { it.focus.reference?.split(TYPE_REFERENCE_DELIMITER)?.firstOrNull() ?: "" }
-              .filterNot { isAppConfig(it.key) }
+              .filter {
+                it.key == ResourceType.Questionnaire.name ||
+                  it.key == ResourceType.StructureMap.name ||
+                  it.key == ResourceType.List.name ||
+                  it.key == ResourceType.PlanDefinition.name ||
+                  it.key == ResourceType.Library.name ||
+                  it.key == ResourceType.Measure.name
+              }
               .forEach { resourceGroup ->
                 val resourceIds =
                   resourceGroup.value.joinToString(",") { sectionComponent ->
                     sectionComponent.focus.extractId()
                   }
                 val searchPath = resourceGroup.key + "?${Composition.SP_RES_ID}=$resourceIds"
-                fhirResourceDataSource.loadData(searchPath).entry.forEach {
-                  addOrUpdate(it.resource)
+                fhirResourceDataSource.loadData(searchPath).entry.forEach { bundleEntryComponent ->
+                  when (bundleEntryComponent.resource) {
+                    is ListResource -> {
+                      addOrUpdate(bundleEntryComponent.resource)
+                      val list = bundleEntryComponent.resource as ListResource
+                      list.entry.forEach { listEntryComponent ->
+                        /*Here we extract the keys and ids for the resources listed in
+                        the List resource */
+                        val resourceKey = listEntryComponent.item.reference.substringBeforeLast("/")
+                        val resourceId = listEntryComponent.item.reference.substringAfterLast("/")
```

Comment:
@ellykits @qaziabubakar can we have a utils function to do this? I assume this is something we will wan to do a lot